src/rules/no-invalid-properties.js

```diff
@@ -124,6 +140,99 @@ export default {
 
 		const [{ allowUnknownVariables }] = context.options;
 
+		/**
+		 * Iteratively resolves CSS variable references until a value is found.
+		 * @param {string} variableName The variable name to resolve
+		 * @param {Map<string, string>} cache Cache for memoization within a single resolution scope
+		 * @param {Set<string>} [seen] Set of already seen variables to detect cycles
+		 * @returns {string|null} The resolved value or null if not found
+		 */
+		function resolveVariable(variableName, cache, seen = new Set()) {
+			/** @type {Array<string>} */
+			const fallbackStack = [];
+			let currentVarName = variableName;
+
+			while (true) {
+				if (seen.has(currentVarName)) {
+					break;
+				}
+				seen.add(currentVarName);
+
+				if (cache.has(currentVarName)) {
+					return cache.get(currentVarName);
+				}
+
+				const valueNode = vars.get(currentVarName);
+				if (!valueNode) {
+					break;
+				}
+
+				const valueText = sourceCode.getText(valueNode).trim();
+				const parsed = parseVarFunction(valueText);
+
+				if (!parsed) {
+					cache.set(currentVarName, valueText);
+					return valueText;
+				}
+
+				if (parsed.fallbackText) {
+					fallbackStack.push(parsed.fallbackText);
+				}
+				currentVarName = parsed.name;
+			}
+
+			while (fallbackStack.length > 0) {
+				const fallbackText = fallbackStack.pop();
+				if (!fallbackText) {
+					continue;
+				}
+				// eslint-disable-next-line no-use-before-define -- resolveFallback and resolveVariable are mutually recursive
+				const resolvedFallback = resolveFallback(
+					fallbackText,
+					cache,
+					seen,
+				);
+				if (resolvedFallback !== null) {
+					return resolvedFallback;
+				}
+			}
+
+			return null;
+		}
+
+		/**
+		 * Resolves a fallback text which can contain nested var() calls.
+		 * Returns the first resolvable value or null if none resolve.
+		 * @param {string} rawFallbackText
+		 * @param {Map<string, string>} cache Cache for memoization within a single resolution scope
+		 * @param {Set<string>} [seen] Set of already seen variables to detect cycles
+		 * @returns {string | null}
+		 */
+		function resolveFallback(rawFallbackText, cache, seen = new Set()) {
+			const trimmedFallback = rawFallbackText.trim();
```

Comment:
The fallback text is already trimmed in `parseVarFunction()`, so this isn't needed here.